web-extensions/chrome/engine/utils.js

```diff
@@ -0,0 +1,66 @@
+/**
+ *
+ * The FrameFinder class is an utility class which finds a frame from the
+ * web-extension based on the provided frame Definition.
+ * All Frame definitions are static members of the class, and this serves as a
+ * central point to reference all frames handled by the web extension.
+ *
+ * @class
+ */
+function FrameFinder() {
+}
+
+FrameFinder.definitions = {
+    SIDEPANEL: {
+        id: 'BmcSidePanel',
+        inspect: frame => !!(frame.showHideSidePanel),
+    },
+};
+
+/**
+ * This static method returns the first frame matching the provided frame
+ * Definition object (which should be one of the frameDefinitions owned by the
+ * class).
+ *
+ * @param {Object} frameDef - the frame Definition as definied by FrameFinder
+ *                            class
+ *
+ * @return {Frame} the DOM frame object matching the definition provided
+ */
+FrameFinder.findWindow = function(frameDef) {
+    if (!frameDef) {
+        alert('Cannot find an frame with unknown Definition');
+    }
+
+    let found = null;
+    if (window === window.top) {
+        // if we're in the top window, CORS will prevent introspecting the
+        // webext-owned frames, but we know the frame' ID
+
+        if (frameDef.id) {
+            found = document.getElementById(frameDef.id).contentWindow;
+            console.warn('FrameFinder.find: Fuond frame with id method');
```

Comment:
Instead of this, return `document.getElementById(frameDef.id).contentWindow;` and put the `console.warn` above the `return`.